schedule/models/events.py

```diff
@@ -59,8 +62,91 @@ def __unicode__(self):
     def get_absolute_url(self):
         return reverse('event', args=[self.id])
 
-    def get_occurrences(self, start, end, skip_booster=False):
+    def get_recent_start_validated(self):
+        """
+         Unserializes and validates self.recent_start. Returns the dictionary
+         represented by self.recent_start if all goes well, or None if there
+         is a validation error.
+         @return recent_start converted to object if it's valid given current
+         state of event; None if it doesn't exist or is invalid.
+        """
+        if self.recent_start == None:
+            return None
+
+        recent_start_data = pickle.loads(str(self.recent_start))
+
+        if (recent_start_data['event']['start'] == self.start and
+              recent_start_data['event']['rule'] == self.rule):
+            return recent_start_data
+        else:
+            return None
+
+
+    @property
+    def recent_occurrence_start(self):
         """
+        A lazy loaded property that ensures we unpickle to get the recent
+        occurrence start only once. Use similar pattern for other properties
+        :return:
+        """
+        if not hasattr(self, '__recent_start'):
+            setattr(self, '__recent_start', self.get_recent_start_validated())
+        recent_start = getattr(self, '__recent_start', None)
+        if (recent_start is not None
+            and 'occurrence' in recent_start
+            and 'start' in recent_start['occurrence']):
+            return recent_start['occurrence']['start']
+        else:
+            return None
+
+
+    def build_recent_start_payload(self, occurrence):
+        """
+        Builds a payload to store in self.recent_start and returns the result
+        @return string representation of payload representing recent start.
+        """
+        event_occurrence_data = {
+            'event': {
+                'start': self.start,
+                'rule': self.rule
```

Comment:
Can change to that assuming self.rule won't ever mutate. Is that a safe assumption (eg. we don't ever go from rule 1 being 'WEEKLY' to 'MONTHLY'